Shopify/cli
# Only fetch dev stores

### WHY are these changes introduced?

When we ask users to choose their dev store, we also show non-dev stores.

### WHAT is this pull request doing?

Only show dev stores instead.

### How to test your changes?

- `spin up partners`
- `yarn create @shopify/app`
- `yarn generate extension`
- `bin/spin yarn shopify app --path <your_newly_created_app`

You should only see dev stores

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
